firefox-ios/Client/Frontend/Library/Bookmarks/Legacy/LegacyBookmarksPanel.swift

```diff
@@ -150,8 +150,9 @@ class LegacyBookmarksPanel: SiteTableViewController,
 
     override func reloadData() {
         viewModel.reloadData { [weak self] in
-            self?.tableView.reloadData()
-
+            DispatchQueue.main.async {
```

Comment:
We usually use 'ensureMainThread' for this kind of scenarios.

Reply:
Interesting - is there any scenario where we should use `DispatchQueue.main.async` over `ensureMainThread`?

Reply:
It is definitely not bad to use `DispatchQueue.main.async`, but we use `ensureMainThread` to check if we are already on the main thread and continue the work synchronously. We explicitly used `DispatchQueue.main.async` in `ToolbarKit` because we didn't have access to `ensureMainThread` there.